spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestCreateActions.java

```diff
@@ -117,47 +119,46 @@ public static Object[][] parameters() {
             "parquet-enabled", "true",
             "cache-enabled",
                 "false" // Spark will delete tables using v1, leaving the cache out of sync
-            )
+            ),
+        "hadoop"
       },
       new Object[] {
         "testhive",
         SparkCatalog.class.getName(),
         ImmutableMap.of(
             "type", "hive",
-            "default-namespace", "default")
+            "default-namespace", "default"),
+        "hive"
       },
       new Object[] {
         "testhadoop",
         SparkCatalog.class.getName(),
         ImmutableMap.of(
             "type", "hadoop",
-            "default-namespace", "default")
+            "default-namespace", "default"),
+        "hadoop"
       }
     };
   }
 
-  @Rule public TemporaryFolder temp = new TemporaryFolder();
-
   private String baseTableName = "baseTable";
   private File tableDir;
   private String tableLocation;
-  private final String type;
-  private final TableCatalog catalog;
 
-  public TestCreateActions(String catalogName, String implementation, Map<String, String> config) {
-    super(catalogName, implementation, config);
-    this.catalog = (TableCatalog) spark.sessionState().catalogManager().catalog(catalogName);
-    this.type = config.get("type");
-  }
+  @Parameter(index = 3)
+  private String type;
 
-  @Before
-  public void before() {
+  private TableCatalog catalog;
+
+  @BeforeEach
+  public void beforeEach() {
```

Comment:
@nastra In your implementation, the constructor in the base class goes under the before() with `@BeforeEach`.
But `@BeforeEach` overrides functions in the subclass which have the same decorator and signature.
So, when I run `TestCreateActions`, the base class before() doesnt run & the environment is not setup. 

For now, I've changed the function name to beforeEach() to highlight the behaviour, but I don't know how to improve it.

Reply:
similar to https://github.com/apache/iceberg/pull/9364 you could just override the method and call super.before()

Reply:
thanks! updated